src/main/java/io/github/externschool/planner/dto/TeacherDTO.java

```diff
@@ -0,0 +1,145 @@
+package io.github.externschool.planner.dto;
+
+import io.github.externschool.planner.entity.profile.Teacher;
+
+import javax.validation.constraints.NotNull;
+import java.util.Objects;
+
+public class TeacherDTO {
+
+    private String verificationKey;
+
+    @NotNull
+    private String firstName;
+
+    @NotNull
+    private String patronymicName;
+
+    @NotNull
+    private String lastName;
+
+    @NotNull
+    private String phoneNumber;
+
+    @NotNull
+    private String email;
+
+    @NotNull
+    private String officer;
+
+    private String schoolSubject;
+
+    public TeacherDTO(){
+
+    }
+
+    public TeacherDTO(String verificationKey, String firstName, String patronymicName, String lastName,
+                      String phoneNumber, String email, String officer, String schoolSubject) {
+        this.verificationKey = verificationKey;
+        this.firstName = firstName;
+        this.patronymicName = patronymicName;
+        this.lastName = lastName;
+        this.phoneNumber = phoneNumber;
+        this.email = email;
+        this.officer = officer;
+        this.schoolSubject = schoolSubject;
+    }
+
+    public Teacher constructTeacher(){
```

Comment:
this method defies single responsibility principle - this is an implementation of a TeacherDTO to Teacher converter, but this class is a data transfer object, not an entity-dto converter. it should be a pojo which is as soon as possible to an appropriate web template, and is used to transfer data from service to web layer
https://www.petrikainulainen.net/software-development/design/understanding-spring-web-application-architecture-the-classic-way